stable/kubernetes-dashboard/templates/deployment.yaml

```diff
@@ -50,7 +50,7 @@ spec:
 {{- end }}        
 {{- if .Values.enableInsecureLogin }}
```

Comment:
Is all this tooling to just make auto-generate-certificates disableable?

Reply:
It probably should not be an else-if at this point. Maybe there should be an extra parameter to explicity enable or disable auto-generate-certificates? Currently it's only creating certificates if protocol Http is `false` (or unset, default is false) or `enable-insecure-login` is set to `true`. This is not right, as it is now impossible to have auto generated certificates when protocol Http is `false` and `enable-insecure-login` is set to `true`. This can happen and won't currently work. It already didn't work either before the change.

Reply:
See what I've done in the 2.0.0 PR to give your opinion :)
Basically, taking your idea and removing the enableInsecureLogin parameter (still allowing to give it through extraArgs), greatly simplifying the logic.